src/python/dxpy/dx_extract_utils/germline_utils.py

```diff
@@ -0,0 +1,331 @@
+from __future__ import annotations
+import json
+import os
+import re
+
+from .filter_to_payload import extract_utils_basepath
+from .input_validation import GENOTYPE_TYPES
+
+
+GENOTYPE_ONLY_TYPES = (
+    "ref",
+    "no-call",
+)
+
+SELECT_LIST_REGEX = r"SELECT\s+(DISTINCT\s+)?(.+?)\s+FROM"
+NAMED_EXPRESSION_REGEX = r"(`?(\w+)?`?\.)?`?(\w+)`?( AS `?(\w+)`?)?"
+
+
+def get_genotype_types(filter_dict):
+    """
+    Returns a list of genotype types for the filter that are use in genotype and allele table queries.
+    """
+
+    genotype_types = filter_dict.get("genotype_type") or GENOTYPE_TYPES
+    return [genotype_type for genotype_type in genotype_types if genotype_type not in GENOTYPE_ONLY_TYPES]
+
+
+def get_genotype_only_types(filter_dict, exclude_refdata, exclude_halfref, exclude_nocall):
+    """
+    Returns a list of genotype types for the filter that are use in genotype table only queries.
+    """
+
+    genotype_type_exclude_flag_map = {
+        "ref": exclude_refdata,
+        "half": exclude_halfref,
+        "no-call": exclude_nocall,
+    }
+    genotype_types = filter_dict.get("genotype_type") or list(genotype_type_exclude_flag_map)
+    genotype_only_types = []
+    for genotype_type, excluded in genotype_type_exclude_flag_map.items():
+        if genotype_type in genotype_types and excluded is not None and not excluded:
+            genotype_only_types.append(genotype_type)
+
+    return genotype_only_types
+
+def get_types_to_filter_out_when_infering(
+    requested_types: list[str],
+) -> list:
+    """
+    Infer option require all genotypes types to be queried.
+    If users wishes to obtain only certain types of genotypes,
+    reminder of the types should be filtered out post querying
+    """
+    if len(requested_types) == 0:
```

Comment:
Minor suggestion to be more Pythonic:

```suggestion
    return [] if not requested_types
```

Reply:
Updated (slightly different), thanks!